contracts/access/Ownable2Step.sol

```diff
@@ -51,7 +51,9 @@ abstract contract Ownable2Step is Ownable {
      */
     function acceptOwnership() public virtual {
         address sender = _msgSender();
-        require(pendingOwner() == sender, "Ownable2Step: caller is not the new owner");
+        if (pendingOwner() != sender) {
+            revert OwnableInvalidOwner(sender);
```

Comment:
The rationale is that the `OwnableUnauthorizedAccount` is fired when the caller is not the owner, which is different to being the `pendingOwner`. Here, not even the `pendingOwner()` is the owner yet.

Also, if an unauthorized account accepts an ownership they can't accept, it should be considered an invalid owner. I agree it's an unauthorized account, so I'd suggest either keeping this error or using the other one but changing its definition.

Which of both sound better to you?

Reply:
> if an unauthorized account accepts an ownership they can't accept, it should be considered an invalid owner

IMO that is an "unauthorized" or "restricted" operation

Reply:
> or using the other one but changing its definition.
Do you mean using `OwnableUnauthorizedAccount` and documenting it as an access violation? That would be good IMO.

Reply:
Answering your comments first:

> My understanding was that OwnableUnauthorizedAccount is emitted when the account is not authorized to do the action

That's correct, however, I'd again mention the contextual information. I tried to add a description to every error to provide more of that information. Particularly for `OwnableUnauthorizedAccount`, the description is: "The caller account is not the owner."

I agree the user is not authorized to do the action, but I think it's important to be consistent with the "**why**" he's not authorized. I have the feeling that if we don't keep consistent with that, the errors will end up being more confusing than useful in the end.

>  If OwnableUnauthorizedAccount is specifically about the caller not being the owner, then I suggest we change the name to something like OwnableOwnerRestricted.

I agree with this change but I'd follow the EIP-6093 syntax and do `OwnableRestrictedOwner` instead.

> IMO that is an "unauthorized" or "restricted" operation

It is. But IMO is as expressive as just throwing `Error`. I strongly believe what is important here is trying to express **why** the owner is restricted. In this case, it's unauthorized because it's an invalid owner, where the validity is defined by not being the pending owner.

> Do you mean using OwnableUnauthorizedAccount and documenting it as an access violation? That would be good IMO.

Correct, we can do that but I still believe the more we generalize an error, the less relevant information it provides. We should be thinking carefully about how much we want to reuse errors deteriorating the meaning of them vs how many hyper-specific errors we want to have.

Reply:
Now, some recap before I formulate my proposal:

- There are 3 errors in `Ownable2Step` (considering the underlying `Ownable` errors)
- I think we both agree on having only 2 errors and share one of them between `Ownable` and `Ownable2Step`
- The error when `pendingOwner() != sender` is an unauthorized operation but it also has to do with the validity of the new owner
- I made a mistake with `OwnableInvalidOwner`, so I'll fix the description of `OwnableInvalidOwner` for something pointing to the validity of the ownership.

My understanding is that we're debating about whether reusing the `OwnableUnauthorizedAccount` or `OwnableInvalidOwner` in the `acceptOwnership`. I have two proposals:

1. Let's use `OwnableUnauthorizedAccount` but update the description to be just `The caller account is not authorized to perform an operation.`
2. Let's keep `OwnableInvalidOwner` with the updated description.

Would you rather go for 1 or 2?

Reply:
I'd go for 1. I believe its both clearer today, and more futur-proof if we ever extend beyound Ownable-2-steps

Reply:
Deal, I just pushed a commit, thanks!